### ISSUE TITLE
tf.keras.backend.repeat_elements does not support negative indexing on tensors with dynamic shape

### ISSUE THREAD

**Issue by josecanedo** [tensorflow/tensorflow#40852]
### System information

-   **Have I written custom code (as opposed to using a stock example script
    provided in TensorFlow)**: yes
-   **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10 and Google Colab
-   **TensorFlow installed from (source or binary)**: binary
-   **TensorFlow version (use command below)**: v2.2.0-0-g2b96f3662b 2.2.0
-   **Python version**: 3.7
-   **CUDA/cuDNN version**: 10.1
-   **GPU model and memory**: Colab GPU

### Describe the problem
The implementation of repeat_elements behave differently whether the input tensor has static or dynamic shape.
For tensors with dynamic shape it does not accept negative indexing for the axis parameter.
For tensors with static shape it accepts negative indexing for the axis parameter.
TensorFlow follow standard python indexing rules. 
There is a workaround using positive indexing.

**How to reproduce**
Run the repeat_elements with axis=-1 and tf.config.experimental_run_functions_eagerly(False)
Note the resulting array
Run with tf.config.experimental_run_functions_eagerly(True)
Note the resulting array
Set axis=1 and repeat 1-4
Note that input_signature parameter in tf.function is there to reproduce a scenario of a graph with the tensor x with dynamic shape.

### Source code / logs

```
import tensorflow as tf
import tensorflow.keras.backend as K
import numpy as np
```
```
@tf.function(input_signature=[tf.TensorSpec(shape=[None, None], dtype=tf.int32)])
def f(x):
  x = K.repeat_elements(x, rep=3, axis=-1)
  return x

tf.config.experimental_run_functions_eagerly(True)
v = tf.Variable([[0, 1],[2, 3]])
f(v)
```
> <tf.Tensor: shape=(2, 6), dtype=int32, numpy=
> array([[0, 0, 0, 1, 1, 1],
>        [2, 2, 2, 3, 3, 3]], dtype=int32)>
```
@tf.function(input_signature=[tf.TensorSpec(shape=[None, None], dtype=tf.int32)])
def f(x):
  x = K.repeat_elements(x, rep=3, axis=-1)
  return x

tf.config.experimental_run_functions_eagerly(False)
v = tf.Variable([[0, 1],[2, 3]])
f(v)
```
> <tf.Tensor: shape=(2, 6), dtype=int32, numpy=
> array([[0, 1, 2, 3, 0, 1],
>        [2, 3, 0, 1, 2, 3]], dtype=int32)>

**Comment by Saduf2019** [comment]
I am able to replicate the reported issue, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/9a76939c1ff63d1b3fb551e0314720c8/untitled244.ipynb)

**Comment by karmel** [comment]
repeat_elements is kept for backwards compatibility right now, but you should prefer the underlying TensorFlow manipulations or [tf.experimental.numpy.repeat](https://www.tensorflow.org/api_docs/python/tf/experimental/numpy/repeat) if you need alternate behavior.
